CustomResourceDefinition/v1alpha2/bool/main.tf

```diff
@@ -0,0 +1,11 @@
+output "schema" {
+  value = {
+    type    = "bool"
+    version = "v1"
+    subItem = null
+    validations = {
+      has_default_value = can(var.manifest.default) ? true : false
+      default_value     = try(var.manifest.default, null)
+    }
+  }
+}
```

Comment:
Senti falta de um `precondition` para o caso de ter um manifesto com `default: null`.

Precisamos validar se o parâmetro passado no manifesto se encaixa com o tipo definido e mostrar uma mensagem de erro bonitinha para quando não for.

Reply:
Qual a ideia nesse caso?
- validar que o valor passado no `default: ` é compatível com o campo do manifesto?
- Validar que não podemos passar `null` como um valor default?

A primeira acho que faz mais sentido. Mas a segundo eu acho que não faria agora pois ela terá unfluência na implementação de nulidade dos campos, que inclusove pode acabar vindo "de graça" depois da implementação de default value pra todo mundo.

Tavez nesse caso aqui, por ser boolean, podemos até impedir um null (pois parece não fazer sentido mesmo), mas para os outros campos eu deixaria em aberto a questão do null.

(vou anotar a validação da compatibilidade de tipos entre o default e o tipo do campo pra fazer depois)

Reply:
Feito. Commits 8f1f89e7d33cec7e01f46c80805ab5be6b1d370b e b53eac895f02f38fd469c4f13e6c726ab93a74e8